docs/architecture/adr-103-proto-versioning.md

```diff
@@ -0,0 +1,140 @@
+# ADR 103: Protobuf definition versioning
+
+## Changelog
+
+- 2023-04-27: First draft (@thanethomson)
+
+## Status
+
+Accepted
+
+## Context
+
+In CometBFT v0.34 through v0.38, the Protocol Buffers definitions in the `proto`
+folder are organized in a relatively flat structure and evolve over time in
+major releases of CometBFT.
+
+For integrators who may want to interact with nodes running different versions
+of CometBFT, this means that, when consuming those Protobuf definitions, they
+need to clone the CometBFT repository and check out the definitions for each and
+every major version they want to support. It also means that they need to
+manually diff the types to get a precise sense of what changed between CometBFT
+releases.
```

Comment:
I would add a sentence saying that, moreover, the differences obtained with the manual diff are often undocumented